server/plugin.go

```diff
@@ -113,13 +120,42 @@ func (p *Plugin) OnActivate() error {
 		return mutateSampleLabels(l, target, *p.configuration.HonorTimestamps, []*relabel.Config{})
 	}
 
+	ticker := time.Tick(time.Duration(scrapeInterval) * time.Second)
+	p.closeChan = make(chan bool)
+
+	go func() {
+		for {
+			select {
+			case <-p.closeChan:
+				return
```

Comment:
We need to signal until this statement has hit. When you run `close(ch)`, it does not guarantee that the receiving end will have received the event. So we need something similar to the websocket closing mechanism. Either another channel, or a sync.WaitGroup.

So on the caller side:

```go
close(ch)
<-closed
```

callee side
```go
case <-p.closeChan:
  closed <- struct{}{}
  return
```

Or the same can be done via a `WaitGroup` as well. 0/5.

Reply:
I think we can ditch this if we use `time.Ticker` and close that while exiting.